core/src/main/kotlin/fr/sncf/osrd/api/api_v2/CommonSchemas.kt

```diff
@@ -30,3 +31,39 @@ class SignalSighting(
     val position: Offset<Path>,
     val state: String,
 )
+
+class RoutingRequirement(
+    val route: String,
+    @Json(name = "begin_time") val beginTime: TimeDelta,
+    val zones: List<RoutingZoneRequirement>
+)
+
+class RoutingZoneRequirement(
+    val zone: String,
+    @Json(name = "entry_detector") val entryDetector: String,
+    @Json(name = "exit_detector") val exitDetector: String,
+    val switches: Map<String, String>,
+    @Json(name = "end_time") val endTime: TimeDelta,
+)
+
+class SpacingRequirement(
+    val zone: String,
+    @Json(name = "begin_time") val beginTime: TimeDelta,
+    @Json(name = "end_time") val endTime: TimeDelta,
+)
+
+/**
+ * Dates can be typed as `LocalDateTime` in json compatible types, with this adapter it will be
+ * converted to/from ISO8601 strings
+ */
+class DateAdapter : JsonAdapter<ZonedDateTime>() {
+    @FromJson
+    override fun fromJson(reader: JsonReader): ZonedDateTime {
+        return ZonedDateTime.parse(reader.nextString())
+    }
+
+    @ToJson
+    override fun toJson(writer: JsonWriter, value: ZonedDateTime?) {
+        writer.value(value?.toString())
+    }
+}
```

Comment:
It should be moved to the file `UnitJsonAdapters`, and can be integrated to `UnitAdapterFactory` so that it doesn't need to be added to each adapter

Reply:
I didn't put it there as it is not really a unit, as Distance, Duration etc. I don't mind putting it there though.